### ISSUE TITLE
The `FIXME` is right, get, pop, setdefault should use the _key function

### ISSUE THREAD

**Issue by karolinepauls** [zx80/cachetools-utils#1]
The fixme: https://github.com/zx80/cachetools-utils/blob/e64587c1f44271c0bf5a23214a7d9fa5bb8f7b7a/CacheToolsUtils.py#L278

Here they are defined: https://github.com/tkem/cachetools/blob/36b83c34afdc27df1a0c34f673b27ba06231c833/src/cachetools/__init__.py#L105

**Comment by zx80** [comment]
Thanks for the pointer!

There is Cachetools' `get` function and Redis `get` function: same name, but not exactly same behavior.

The `FIXME` comment was about the later: if someone is using the `RedisCache` class as a replacement for `redis.Cache`, then probably they already took care of the key constraints, so that doing it again would be a bad idea?

But from Cachetools' `get` perspective it should be done, indeed.

:confused: I'm still unclear. Maybe a constructor option would make sense?

**Comment by zx80** [comment]
As for cachetools' own `get` function, I do not attempt to implement it, but probably I should, and this one *should* indeed filter the key so that it behaves like `[]`.

**Comment by karolinepauls** [comment]
Here you can see an example of conusion between prefixed and unprefixed keys.

```
In [17]: from redis import StrictRedis

In [18]: import CacheToolsUtils as ctu

In [19]: client = StrictRedis(host='redis', db=1)

In [20]: cache = ctu.PrefixedRedisCache(cache=client, prefix='PREFIX_', ttl=100000)

In [21]: cache.set('a', 1)
Out[21]: True

In [22]: cache.get('a')
Out[22]: b'1'

In [23]: cache['a'] = 1

In [24]: cache['a']
Out[24]: 1

In [25]: cache.get('a')
Out[25]: b'1'
```

Redis commands sent by the session above:

```
$ redis-cli monitor | grep -v '"ping"'
OK
1662553377.675058 [0 172.20.0.5:38306] "SELECT" "1"
1662553377.675290 [1 172.20.0.5:38306] "SET" "a" "1"
1662553384.671844 [1 172.20.0.5:38306] "GET" "a"
1662553395.506427 [1 172.20.0.5:38306] "SET" "PREFIX_a" "1" "EX" "100000"
1662553401.242975 [1 172.20.0.5:38306] "GET" "PREFIX_a"
1662553407.753768 [1 172.20.0.5:38306] "GET" "a"
```

I don't really understand the point of implementing methods like `get` - `colections.abc.MutableMapping` should already implement them, and all you need to supply are a couple of dunder methods - see https://github.com/python/cpython/blob/83886261fa584b45d5e70c5a422f69cad7422a0a/Lib/_collections_abc.py#L883-L885.

**Comment by zx80** [comment]
> I don't really understand the point of implementing methods like get - colections.abc.MutableMapping should already implement them, and all you need to supply are a couple of dunder methods - see https://github.com/python/cpython/blob/83886261fa584b45d5e70c5a422f69cad7422a0a/Lib/_collections_abc.py#L883-L885.

Cachetools does implement these in its base class, as you pointed out, so naïvely I'd reimplement them as well, but indeed they are already implemented in the hierarchy so it is totally useless.

**Comment by zx80** [comment]
Fixed in v4.3, hopefully.
